crates/nix_rs/src/flake/outputs.rs

```diff
@@ -68,25 +68,86 @@ impl FlakeOutputs {
     }
 }
 
+/// Represents a leaf value of a flake output
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Leaf {
+    Val(Val),
+    Unknown(Unknown),
+    Filtered(Filtered),
+    Skipped(Skipped),
+    Doc(String),
+}
+
+impl Leaf {
+    /// Get the value as a [Val]
+    pub fn as_val(&self) -> Option<&Val> {
+        match self {
+            Self::Val(v) => Some(v),
+            _ => None,
+        }
+    }
+}
+
 /// The metadata of a flake output value which is of non-attrset [Type]
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Val {
-    #[serde(rename = "type")]
+    #[serde(rename = "what")]
     pub type_: Type,
-    pub name: Option<String>,
-    pub description: Option<String>,
+    pub derivation_name: Option<String>,
+    pub short_description: Option<String>,
+}
+
+impl Default for Val {
+    fn default() -> Self {
+        Self {
+            type_: Type::Unknown,
+            derivation_name: None,
+            short_description: None,
+        }
+    }
+}
+
+/// Boolean flags at the leaf of a flake output
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Unknown {
+    pub unknown: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename = "camelCase")]
+pub struct Filtered {
+    pub filtered: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Skipped {
+    pub skipped: bool,
 }
 
 /// The type of a flake output [Val]
 ///
 /// [Nix source ref](https://github.com/NixOS/nix/blob/2.14.1/src/nix/flake.cc#L1105)
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
 pub enum Type {
+    #[serde(rename = "NixOS module")]
     NixosModule,
-    Derivation,
+    #[serde(rename = "NixOS configuration")]
+    NixosConfiguration,
+    #[serde(rename = "nix-darwin configuration")]
+    DarwinConfiguration,
+    #[serde(rename = "package")]
+    Package,
+    #[serde(rename = "development environment")]
+    DevShell,
+    #[serde(rename = "CI test")]
+    Check,
+    #[serde(rename = "app")]
```

Comment:
Are we actually dealing with a JSON that uses human-readable description (rather inconsistently at that, cf. `package`) as object *keys*?

Reply:
Right. So can you update the `Type` struct comment? Because it is out of date.

<img width="291" alt="image" src="https://github.com/user-attachments/assets/7c2a09ad-d926-4203-be2f-5d5e4ae1eba4">

Reply:
Do you mean this:

> /// \[Nix source ref\](https://github.com/NixOS/nix/blob/2.14.1/src/nix/flake.cc#L1105)

Reply:
If yes, should I point it to https://github.com/DeterminateSystems/flake-schemas/blob/0a5c42297d870156d9c57d8f99e476b738dcd982/flake.nix#L253

Reply:
Yes

Reply:
https://github.com/juspay/omnix/pull/166/commits/ce9d94f67993fa1b4d4d964f113e08ddf7a6ea77